addons/hr_expense/models/hr_expense.py

```diff
@@ -542,6 +542,16 @@ def write(self, vals):
         if 'tax_ids' in vals or 'analytic_distribution' in vals or 'account_id' in vals:
             if any(not expense.is_editable for expense in self):
                 raise UserError(_('You are not authorized to edit this expense report.'))
+
+        enforced_non_zero_expenses = self if 'sheet_id' in vals else self.filtered('sheet_id')
```

Comment:
I think we also need to check the value because If the value is set to false or None, we shouldn't check the amounts.
```suggestion
        if 'sheet_id' in vals:
            if vals['sheet_id']:
                to_checkenforced_non_zero_expenses = self
            else:
                to_checkenforced_non_zero_expenses  = False
        else:
            to_checkenforced_non_zero_expenses = self.filtered('sheet_id')
```